update CONTRIBUTING (#4180)
tendermint/tendermint: CONTRIBUTING.md

# Contributing

Thank you for your interest in contributing to tendermint!  Before
contributing, it may be helpful to understand the goal of the project. The goal

> of tendermint is to develop a BFT consensus engine sufficiently robust to

Reviewer: "sufficiently" is a bit vague term imo

> of tendermint is to develop a BFT consensus engine sufficiently robust to

Reviewer 2: What do you think about this formulation: "The goal of tendermint is to develop a planetary scale, correct, secure and reliable BFT consensus engine, that should facilitate development of permisionless value carrying networks."

> of tendermint is to develop a BFT consensus engine sufficiently robust to

Reviewer 3: ```suggestion
of Tendermint is to develop a BFT consensus engine sufficiently robust to
```

maybe? I see it's lowercase everywhere in this guide, but I didn't know we style it that way

> support permissionless value carrying networks.  While all contributions are

Reviewer: > permissionless value carrying networks

do you mean large networks (> 300 vals)? because some might say TM is already sufficiently robust for networks < 100 validators

Author: Not sure if the distinction here changes the interpretation of the text. Said another way, I'm not sure the distinction improves the vagueness of the sentence 🤣

> support permissionless value carrying networks.  While all contributions are

Reviewer 3: ```suggestion
support permissionless value-carrying networks.  While all contributions are
```

welcome, contributors should bear this goal in mind in deciding if they should
target the main tendermint project or a potential fork.  When targeting the
main tendermint project, the following process leads to the best chance of
landing changes in master.

> All contributions should start with an Issue. The issue helps capture the

Reviewer 3: ```suggestion
All contributions should start with a Github Issue. The issue helps capture the
```

> problem your trying to solve and allows early feedback.  Once the problem

Reviewer 3: ```suggestion
problem you're trying to solve and allows for early feedback.  Once the problem
```

description has been captured by an issue, the process can proceed in different

> direction depending on how well defined the problem and proposed

Reviewer 3: ```suggestion
directions depending on how well defined the problem and proposed
```

> solution are.  If the change is simple and well understood, maintainers will +1

Reviewer 3: What does it mean to +1 an issue? We don't use Gerrit, so maybe we should define this in Github-oriented terms

> the issue and sign off on reviewing such a change when it’s ready Changes

Reviewer 3: ```suggestion
the issue and sign off on reviewing such a change when it’s ready. Changes
```

should start with a [draft pull](https://github.blog/2019-02-14-introducing-draft-pull-requests/) request against master. The draft signals that

> work is underway, even if it’s not ready for review When changes are ready for

Reviewer 3: ```suggestion
work is underway, even if it’s not ready for review. When changes are ready for
```

Reviewer 3: Should people also expect conversation to happen on a draft PR? e.g. is a draft PR something that people can/will review? Or is it just a placeholder?

> review, they should be kept up to date with master If the issue is well

Reviewer 3: When does a draft PR become a non-draft PR?

understood but leads to some structural change to the codebase, any change

> should be preceded by an [Architectural Decision Record (ADR)](./docs/architecture/). The ADR will help build consensus on an overall

Reviewer: I think we also should tell how one can reserve a number for an ADR

strategy to changes to ensure the code base maintains coherence in the larger
context.  If you are not comfortable with writing an ADR, you can open a
less-formal issue and the Tendermint team will help you turn it into an ADR.
If the issue would benefit from thorough discussion, maintainers of tendermint
may request that you create an [Request For Comment](https://github.com/tendermint/spec/tree/master/rfc). Discussion at the RFC stage
will help build the consensus necessary to ensure changes land in the protocol
in a timely fashion.  If the conclusion of the discussion is a large structural
change to the codebase, an ADR should follow with a proposed strategy.

> 

Reviewer 3: Maybe also mention that for problems that require an RFC (or even a more complex ADR), we may want to have conversations outside of Github, whether that's through the Tendermint Friends Slack or through a dev call?

# Contributing

Thank you for your interest in contributing to Tendermint! Before
contributing, it may be helpful to understand the goal of the project. The goal
of Tendermint is to develop a BFT consensus engine robust enough to
support permissionless value-carrying networks. While all contributions are
welcome, contributors should bear this goal in mind in deciding if they should
target the main tendermint project or a potential fork. When targeting the
main Tendermint project, the following process leads to the best chance of
landing changes in master.

> All contributions should start with a [Github

Reviewer: "If you would like to work on the issue which already exists, please indicate so by leaving a comment."

welcome, contributors should bear this goal in mind in deciding if they should
target the main tendermint project or a potential fork. When targeting the
main Tendermint project, the following process leads to the best chance of
landing changes in master.

All work on the code base should be captured by a [Github
Issue](https://github.com/tendermint/tendermint/issues).
[Search](https://github.com/tendermint/tendermint/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)
is a good place start when looking for places to contribute.  If you
would like to work on an issue which already exists, please indicate so
by leaving a comment.

All new contributions should start with a [Github
Issue](https://github.com/tendermint/tendermint/issues). The issue helps
capture the problem you're trying to solve and allows for early
feedback. Once the problem has been captured by an issue, the process
can proceed in different directions depending on how well defined the
problem and potential solution are. If the change is simple and well
understood, maintainers will indicate their support with a heartfelt 
emoji.

If the issue would benefit from thorough discussion, maintainers may

> request that you create a [Request For

Reviewer 4: Are Rfc's merged into the repo or discarded when it's understood?
